src/PlaywrightSharp.Firefox/FirefoxProcessManager.cs

```diff
@@ -0,0 +1,512 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using PlaywrightSharp.Helpers;
+
+namespace PlaywrightSharp.Firefox
+{
+    internal class FirefoxProcessManager : IProcessManager

```

Comment:
@Meir017 do you think we can reuse this `ProcessManager` among browsers?

Reply:
I'm sure we can. we will need to add some abstraction for that. firefox gets the ws url from the output data and not the error data but it's makes sense to use the same logic

Reply:
I will create an issue for that, we will have a better picture when we implement Webkit.